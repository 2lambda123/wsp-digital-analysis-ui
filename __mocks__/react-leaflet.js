/* global jest */

import React from 'react'

const ReactLeaflet = require.requireActual('react-leaflet')

// mock tile layer urls
process.env.LEAFLET_TILE_URL = 'mock.url/tile'

ReactLeaflet.GeoJSON.prototype.createLeafletElement = function () {
  return {
    data: this.props.data,
    getLayers () {
      return []
    }
  }
}

ReactLeaflet.GeoJSON.prototype.render = function () {
  return (
    <g type='GeoJSON' data={this.props.data}>
      {this.props.children}
    </g>
  )
}

ReactLeaflet.Marker.prototype.render = function () {
  return (
    <g type='Marker' {...this.props}>
      {this.props.children}
    </g>
  )
}

ReactLeaflet.Path.prototype.setStyle = jest.fn()

module.exports = ReactLeaflet
