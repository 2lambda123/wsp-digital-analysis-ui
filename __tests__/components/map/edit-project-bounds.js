/* global describe, it, expect, jest */

import { mount } from 'enzyme'
import Leaflet from '../../../test-utils/mock-leaflet'
import React from 'react'
import { Map } from 'react-leaflet'

import EditProjectBounds from '../../../lib/components/map/edit-project-bounds'

describe('Components > Map > EditProjectBounds', () => {
  it('renders correctly', () => {
    const props = {
      bounds: {
        north: 12,
        east: -34,
        south: -34,
        west: 12
      },
      isLoaded: true,
      saveLocally: jest.fn()
    }

    const tree = mount(
      <Map>
        <EditProjectBounds
          {...props}
          />
      </Map>
    , {
      attachTo: document.getElementById('test')
    })
    expect(Leaflet.marker.mock.calls[0][0]).toMatchSnapshot()
    expect(Leaflet.marker.mock.calls[1][0]).toMatchSnapshot()
    expect(Leaflet.rectangle.mock.calls[0][0]).toMatchSnapshot()
    const mountedMap = tree.find(Map)
    expect(mountedMap.node.leafletElement.getBounds()).toMatchSnapshot()
  })
})
