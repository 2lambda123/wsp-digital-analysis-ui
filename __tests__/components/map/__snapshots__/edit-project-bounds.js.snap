exports[`Components > Map > EditProjectBounds renders correctly 1`] = `
Array [
  -34,
  12,
]
`;

exports[`Components > Map > EditProjectBounds renders correctly 2`] = `
Array [
  12,
  -34,
]
`;

exports[`Components > Map > EditProjectBounds renders correctly 3`] = `
Array [
  Array [
    -34,
    12,
  ],
  Array [
    12,
    -34,
  ],
]
`;

exports[`Components > Map > EditProjectBounds renders correctly 4`] = `
Array [
  Array [
    -34,
    12,
  ],
  Array [
    12,
    -34,
  ],
]
`;
